pkg/controllers/common/utils.go

```diff
@@ -131,7 +131,7 @@ func GetVirtualMachineNameForSubnetPort(subnetPort *v1alpha1.SubnetPort) (string
 		return "", nil
 	}
 	array := strings.Split(attachmentRef, "/")
-	if len(array) != 2 || !strings.EqualFold(array[0], servicecommon.ResourceTypeVirtualMachine) {
+	if len(array) != 3 || !strings.EqualFold(array[0], servicecommon.ResourceTypeVirtualMachine) {
 		err := fmt.Errorf("invalid annotation value of '%s': %s", servicecommon.AnnotationAttachmentRef, attachmentRef)
 		return "", err
 	}
```

Comment:
Will you have another patch to put interface name in store?

Reply:
This PR is to unblock integration test. Other changes will be added to controller PR.

Reply:
Currently nsx-operator doesn't have extra indexer for K8s store. I checked vmMapFunc and it listed all SubnetPorts to find proper VM name, and I'm not sure how to add K8s store indexer into current nsx-operator.